server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -265,6 +265,11 @@ static boolean execute(SearchContext searchContext,
                 searcher.search(query, queryCollector);
             } catch (TimeExceededException e) {
                 assert timeoutSet : "TimeExceededException thrown even though timeout wasn't set";
+                
+                if (!searchContext.request().allowPartialSearchResults()) {
```

Comment:
nit: I think the code style is to avoid negating using `!X` and prefer `X == false` so its harder to miss the negation